### PR TITLE
feat: add support for specifying date format

### PR DESCRIPTION
**Pull request by JessicaLHartog**
<details open>
<summary><h2> Problem </h2></summary>

Customers with an on-premise masking engine are able to specify custom date formats in their rulesets so that Date Shift algorithms (among others) can correctly understand how to interpret dates. Our Azure templates don't currently support this feature.
</details>

<details open>
<summary><h2> Solution </h2></summary>

Create a pathway for customers to specify, and the masking pipeline to respect, custom date formats.
</details>

<details>
<summary><h2> Testing Done </h2></summary>

Built a release, imported the pipeline, modified the metadata for some fields to specify the date format, and re-ran the masking pipeline.
</details>

<details open>
<summary><h2> Notes to Reviewers </h2></summary>

Also updated the descriptions provided into the `Description` boxes of the dataflow steps in the templates as well, and renamed dataflow elements where appropriate so that they are consistent with documentation added in #12 as much as possible.
</details>

### CODE REVIEW

**Review by nhlien93 — APPROVED** [review]

File: dcsazure_Snowflake_to_Snowflake_mask_pl/dcsazure_Snowflake_to_Snowflake_mask_pl.json [hunk 1]
```diff
@@ -196,7 +196,7 @@
                           "type": "Expression"
                         },
                         "DF_FIELD_ALGORITHM_ASSIGNMENT": {
-                          "value": "'@{activity('Get Masking Parameters').output.runStatus.output.MaskingParameterOutput.value[0].FieldAlgorithmAssignments}'",
+                          "value": "'@{activity('Get Masking Parameters').output.runStatus.output.MaskingParameterOutput.value[0].FieldAlgorithmAssignment}'",
```
Comment (pilarpickering) [inline dcsazure_Snowflake_to_Snowflake_mask_pl/dcsazure_Snowflake_to_Snowflake_mask_pl.json:199]: I didn't load the template yet, but maybe you don't need this change. I think the output has an `s`

see line 715 below.
```
FieldAlgorithmAssignments = regexReplace(reduce(mapAssociation(keyValues(collect(identified_column),
```

File: dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_prof_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_prof_pl.json [hunk 2]
```diff
@@ -485,15 +484,16 @@
...
+              "description": "Import the first million rows of unmasked data from the desired source dataset"
             },
             {
               "linkedService": {
                 "referenceName": "[parameters('Metadata Datastore')]",
                 "type": "LinkedServiceReference"
               },
-              "name": "MetadataStoreRead"
+              "name": "MetadataStoreRead",
+              "description": "Import data from Metadata Datastore, querying the ruleset table"
```
Comment (pilarpickering) [inline dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_prof_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_prof_pl.json:496]: wow, thank you for adding all these individual descriptions!

File: dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl.json [hunk 3]
```diff
@@ -595,51 +599,92 @@
...
+            },
+            {
+              "name": "RulesetWithTypes",
+              "description": "Join the ruleset table with the type mapping table based on the type of the column and the translation of that type to an ADF type"
+            },
+            {
+              "name": "RulesetWithAlgorithmTypeMapping",
+              "description": "Generate several columns:\n\n    output_row that always contains 1 (used later for Aggregate and Join operations)\n    adf_type_conversion that contains a string like <column_name> as <adf_type>\n    column_width_estimate that contains an integer that uses DF_COLUMN_WIDTH_ESTIMATE as the width for any column where identified_column_max_length is not positive, and identified_column_max_length plus some padding otherwise"
```
Comment (pilarpickering) [inline dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl.json:631]: Just curious, does the UI resolve the `\n`?

**Review by pilarpickering — APPROVED** [review]
Thank you!

File: dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl.json (hunk 3 above)
Comment (JessicaLHartog) [reply dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl/dcsazure_Snowflake_Legacy_to_Snowflake_Legacy_mask_pl.json:631]: It does :)

**Pull request merged**